code/modules/mob/living/simple_animal/hostile/retaliate/icarus_drone.dm

```diff
@@ -0,0 +1,272 @@
+			step_to(O, get_turf(pick(view(7, src))))
+		if(prob(25))
+			O = new /obj/item/stack/rods(get_turf(src))
+			step_to(O, get_turf(pick(view(7, src))))
+
+		//plasteel
+		O = new /obj/item/stack/material/plasteel(get_turf(src))
+		step_to(O, get_turf(pick(view(7, src))))
+		if(prob(75))
+			O = new /obj/item/stack/material/plasteel(get_turf(src))
+			step_to(O, get_turf(pick(view(7, src))))
+		if(prob(50))
+			O = new /obj/item/stack/material/plasteel(get_turf(src))
+			step_to(O, get_turf(pick(view(7, src))))
+		if(prob(25))
+			O = new /obj/item/stack/material/plasteel(get_turf(src))
+			step_to(O, get_turf(pick(view(7, src))))
+
+		//also drop dummy circuit boards deconstructable for research (loot)
+		var/obj/item/circuitboard/C
+
+		//spawn 1-4 boards of a random type
+		var/spawnees = 0
+		var/num_boards = rand(1, 4)
+		var/list/options = list(1, 2, 4, 8, 16, 32, 64, 128, 256)
+		for(var/i = 0, i < num_boards, i++)
+			var/chosen = pick(options)
+			options.Remove(options.Find(chosen))
+			spawnees |= chosen
+
+		if(spawnees & 1)
+			C = new(get_turf(src))
+			C.name = "drone CPU motherboard"
+			C.origin_tech = list(TECH_DATA = rand(3, 6))
+
+		if(spawnees & 2)
+			C = new(get_turf(src))
+			C.name = "drone neural interface"
+			C.origin_tech = list(TECH_BIO = rand(3, 6))
+
+		if(spawnees & 4)
+			C = new(get_turf(src))
+			C.name = "drone suspension processor"
+			C.origin_tech = list(TECH_MAGNET = rand(3, 6))
+
+		if(spawnees & 8)
+			C = new(get_turf(src))
+			C.name = "drone shielding controller"
+			C.origin_tech = list(TECH_BLUESPACE = rand(3, 6))
+
+		if(spawnees & 16)
+			C = new(get_turf(src))
+			C.name = "drone power capacitor"
+			C.origin_tech = list(TECH_POWER = rand(3, 6))
+
+		if(spawnees & 32)
+			C = new(get_turf(src))
+			C.name = "drone hull reinforcer"
+			C.origin_tech = list(TECH_MATERIAL = rand(3, 6))
+
+		if(spawnees & 64)
+			C = new(get_turf(src))
+			C.name = "drone auto-repair system"
+			C.origin_tech = list(TECH_ENGINEERING = rand(3, 6))
+
+		if(spawnees & 128)
+			C = new(get_turf(src))
+			C.name = "drone phoron overcharge counter"
+			C.origin_tech = list(TECH_PHORON = rand(3, 6))
+
+		if(spawnees & 256)
+			C = new(get_turf(src))
+			C.name = "drone targetting circuitboard"
+			C.origin_tech = list(TECH_COMBAT = rand(3, 6))
+
+	return ..()
+
+/mob/living/simple_animal/hostile/retaliate/icarus_drone/validator_living(var/mob/living/L, var/atom/current)
+	if(L.health <= 0)
+		return FALSE
+	if(istype(L, /mob/living/simple_animal/hostile/hivebot))
+		return TRUE
+	if(istype(L, /mob/living/simple_animal/hostile/carp))
+		return TRUE
+	if(istype(L, /mob/living/simple_animal/hostile/retaliate/cavern_dweller))
+		return TRUE
+	if(istype(L, /mob/living/simple_animal/hostile/giant_spider))
+		return TRUE
+	if(istype(L, /mob/living/simple_animal/hostile/spider_queen))
+		return TRUE
+	if(istype(L, /mob/living/simple_animal/hostile/retaliate/malf_drone))
+		return TRUE
+	return FALSE
```

Comment:
You should probably toss all these mobs in a list and check is_type_in_list(L, listname).
That might actually be a good thing to slap on the mob/blah/hostile parent for fine-tuning future mob targeting? And set the drone's list to all these things. If you feel like it.

Reply:
alright